emr-dynamodb-hive/src/main/java/org/apache/hadoop/hive/dynamodb/DynamoDBStorageHandler.java

```diff
@@ -169,13 +169,19 @@ public void configureTableJobProperties(TableDesc tableDesc, Map<String, String>
             .getProperty(DynamoDBConstants.THROUGHPUT_WRITE_PERCENT));
       }
 
-      String readThroughput = description.getProvisionedThroughput().getReadCapacityUnits()
-          .toString();
-      String writeThroughput = description.getProvisionedThroughput().getWriteCapacityUnits()
-          .toString();
+      if(description.getBillingModeSummary().getBillingMode().equals(DynamoDBConstants.BILLING_MODE_PROVISIONED)) {
+        jobProperties.put(DynamoDBConstants.READ_THROUGHPUT, description.getProvisionedThroughput().getReadCapacityUnits()
+                .toString());
+        jobProperties.put(DynamoDBConstants.WRITE_THROUGHPUT,description.getProvisionedThroughput().getWriteCapacityUnits()
+                .toString());
+      }
```

Comment:
[minor] some style issue, I think the committer will take care of this indentation and link break for you